qualtran/bloqs/arithmetic.py

```diff
@@ -444,11 +446,132 @@ def signature(self):
     def pretty_name(self) -> str:
         return "a gt b"
 
-    def t_complexity(self):
+    def t_complexity(self) -> 'TComplexity':
+        return t_complexity(LessThanEqualGate(self.bitsize, self.bitsize))
+
+    def bloq_counts(
+        self, ssa: Optional['SympySymbolAllocator'] = None
+    ) -> Set[Tuple[Union[int, sympy.Expr], Bloq]]:
         # TODO Determine precise clifford count and/or ignore.
         # See: https://github.com/quantumlib/cirq-qubitization/issues/219
         # See: https://github.com/quantumlib/cirq-qubitization/issues/217
-        return TComplexity(t=8 * self.bitsize)
+        t_complexity = self.t_complexity()
+        return {(t_complexity.t, TGate())}
 
-    def bloq_counts(self, ssa: Optional['SympySymbolAllocator'] = None) -> Set[Tuple[int, Bloq]]:
-        return {(8 * self.bitsize, TGate())}
+
+@frozen
+class GreaterThanConstant(Bloq):
+    r"""Implements $U_a|x\rangle = U_a|x\rangle|z\rangle = |x\rangle |z ^ (x > a)\rangle"
+
+    Args:
+        bitsize: bitsize of x register.
+        val: integer to compare x against (a above.)
+
+    Registers:
+        x: Register to compare against val.
+        result: Register to hold result of comparison.
+    """
+
+    bitsize: int
+    val: int
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build(x=self.bitsize, result=1)
+
+    def t_complexity(self) -> TComplexity:
+        return t_complexity(LessThanGate(self.bitsize, val=self.val))
+
+    def bloq_counts(
+        self, ssa: Optional['SympySymbolAllocator'] = None
+    ) -> Set[Tuple[Union[int, sympy.Expr], Bloq]]:
+        # TODO Determine precise clifford count and/or ignore.
+        # See: https://github.com/quantumlib/cirq-qubitization/issues/219
+        # See: https://github.com/quantumlib/cirq-qubitization/issues/217
+        t_complexity = self.t_complexity()
+        return {(t_complexity.t, TGate())}
+
+
+@frozen
+class EqualsAConstant(Bloq):
+    r"""Implements $U_a|x\rangle = U_a|x\rangle|z\rangle = |x\rangle |z ^ (x == a)\rangle"
+
+    Args:
+        bitsize: bitsize of x register.
+        val: integer to compare x against (a above.)
+
+    Registers:
+        x: Register to compare against val.
+        result: Register to hold result of comparison.
+    """
+
+    bitsize: int
+    val: int
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build(x=self.bitsize, result=1)
+
+    def t_complexity(self) -> 'TComplexity':
+        return LessThanGate(self.bitsize, val=self.val).t_complexity()
```

Comment:
`LessThanGate` is not the correct one here. Equality is always $4 (bitsize - 1)$ regardlress of whether it's between a register and a constant or 2 registers. see https://qualtran.readthedocs.io/en/latest/bloqs/comparison_gates.html#equality-as-a-special-case

Reply:
Done.